do not persist sourcePosition of resources

filip-zyzniewski opened the pull request:
It causes big state diffs if ran in a different
environment (venv path, CI vs local etc.).

It's a reference to the location of the SDK code, not the Pulumi program code, so it's not very useful.

Pulumi seems happy without it.

filip-zyzniewski committed:
do not persist sourcePosition of resources

It causes big state diffs if ran in a different
environment (venv path, CI vs local etc.).

It's a reference to the location of the SDK code, not the
Pulumi program code, so it's not very useful.

Pulumi seems happy without it.

pbzdyl commented on utilities/pulumi_state_splitter/tests/test_split.py:
```diff
@@ -197,6 +204,7 @@ def test_load(self):
         for state in state_dir.state, want:
             state.checkpoint.latest.resources.sort(key=util.resource_key)
             for resource in state.checkpoint.latest.resources:
+                _maybe_delattr(resource, "sourcePosition")
```
**Optional nit:** "maybe delete" sounds a bit vague (under what conditions it will be removed)? Maybe `_delattr_if_exists` or `_safe_delattr`?

filip-zyzniewski replied:
Done in 2d3118369d42876dd02019335b50373c0246299f .

pbzdyl commented on utilities/pulumi_state_splitter/pulumi_state_splitter/model.py:
```diff
@@ -22,7 +22,10 @@ class Resource(pydantic.BaseModel):
-    file_exclude: ClassVar = {"parent_resource"}
+    file_exclude: ClassVar = {
+        "parent_resource",
+        "sourcePosition",
```
**Suggestion:** Could we add a comment documenting the reason for removing this entry?

filip-zyzniewski replied:
Done in d992dd5f484c5ce83f741f05c95193387e6d5661 .

filip-zyzniewski committed:
test_split: s/_maybe_delattr/_delattr_if_exists/

Per
https://github.com/house-reliability-engineering/pulumi/pull/4#discussion_r1464749906

filip-zyzniewski committed:
model: comment on excluding Resource.sourcePosition

Per
https://github.com/house-reliability-engineering/pulumi/pull/4#discussion_r1464750632
